### PR TITLE
fix(ComparableScyllaVersion): support old non-semver versions

### PR DESCRIPTION
**Pull request by fruch**
there are case we might need to use very old verison like `2022.1-rc8` which are not semver complient, and we need to align it before using it

### Testing

- [ ]

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevent to this change (if needed)

### CODE REVIEW

File: unit_tests/test_version_utils.py [hunk 1]
```diff
@@ -450,6 +450,8 @@ def raise_request_error(**kwargs):
     ("5.1.1", (5, 1, 1, '', '')),
     ("5.1.0-rc1", (5, 1, 0, 'rc1', '')),
     ("5.1.0~rc1", (5, 1, 0, 'rc1', '')),
+    ("2022.1-rc8", (2022, 1, 0, 'rc8', '')),
+    ("2022.1~rc8", (2022, 1, 0, 'rc8', '')),
```
Comment (vponomaryov) [inline unit_tests/test_version_utils.py:454]: The same line is added twice.

File: unit_tests/test_version_utils.py [hunk 2]
```diff
@@ -461,6 +463,7 @@ def raise_request_error(**kwargs):
     ("5.2.0-dev-0.20230109.08b3a9c786d9-aarch64", (5, 2, 0, "dev-0.20230109", "08b3a9c786d9")),
     ("2024.2.0.dev.0.20231219.c7cdb16538f2.1", (2024, 2, 0, "dev-0.20231219", "c7cdb16538f2.1")),
     ("2024.1.0.rc2.0.20231218.a063c2c16185.1", (2024, 1, 0, "rc2-0.20231218", "a063c2c16185.1")),
+
```
Comment (vponomaryov) [inline unit_tests/test_version_utils.py:466]: Redundant change

File: sdcm/utils/version_utils.py [hunk 3]
```diff
@@ -146,7 +146,11 @@ def parse(version_string: str):
 
         # NOTE: make short scylla version like '5.2' be correct semver string
         _scylla_version_parts = _scylla_version.split('.')
-        if len(_scylla_version_parts) == 2:
+        if '-' in _scylla_version_parts[1]:
```
Comment (vponomaryov) [inline sdcm/utils/version_utils.py:149]: Unit tests fail on the single-element input values such as `2023`:
```
15:01:28  >       if '-' in _scylla_version_parts[1]:
15:01:28  E       IndexError: list index out of range
```

So, need to check that second element does exist.

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: unit_tests/test_version_utils.py (hunk 2 above)
Comment (fruch) [reply unit_tests/test_version_utils.py:466]: removed

File: sdcm/utils/version_utils.py (hunk 3 above)
Comment (fruch) [reply sdcm/utils/version_utils.py]: fixed

**Pull request merged**